nightingale_interface/main.py

```diff
@@ -9,59 +11,94 @@
 from kivymd.app import MDApp
 from kivy.uix.screenmanager import ScreenManager
 import kivy
-from kivy.properties import NumericProperty
+from kivy.properties import NumericProperty, StringProperty
 
 import MovoConfig
 from screen_wrapper import ScreenWrapper
-from screens.screen_config import ScreenConfig as cfg
+
+from screens.screen_config import ScreenConfig
+from nightingale_ros_bridge.src.nightingale_ros_bridge.bridge_interface_config import (
+    RobotStatus,
+    BridgeConfig,
+    UserInputs,
+)
 
 
 class MainApp(MDApp, ScreenWrapper):
     _other_task = None
-    coms_enabled = False
+    _wd_task = None
+    coms_enabled = True
 
     # ros things
     client = None
-    dummy_topic = None
+    ros_action_topic = None
+    interface_screen_topic = None
+    estop_topic = None
 
     # task queue things
-    queue_task = []
-    queue_delay = []
+    task_queue = []  # dict of { "task": Int, "delay": Int, "args":""}
 
     # counters for items
     water_count = NumericProperty(0)
     ice_count = NumericProperty(0)
     blanket_count = NumericProperty(0)
 
+    ros_set_screen = ""
+
+    watchdog_timer = ScreenConfig.WATCHDOG_TIMER_SECONDS
+    watchdog2_exited = False
+
+    # stack for screens to be able to return to them
+    screen_stack = []
+    current_action = "0"
+    pending_action = "0"
+    payload = ""
+
     def main(self):
         self._other_task = asyncio.ensure_future(self.backend())
+        self._wd_task = asyncio.ensure_future(self.watchdog_run())
 
         async def run_wrapper():
             await self.async_run(async_lib="asyncio")
             print("App done")
             self._other_task.cancel()
+            self._wd_task.cancel()
 
-        return asyncio.gather(run_wrapper(), self._other_task)
+        return asyncio.gather(run_wrapper(), self._other_task, self._wd_task)
 
     # add a task to do after a delay
-    def queue(self, task, delay=0):
-        self.queue_delay.append(delay)
-        self.queue_task.append(task)
+    def queue(self, task, delay=0, args=None):
+        self.task_queue.append({"task": task, "delay": delay, "args": args})
 
     # backend polling to read from ros coms
     async def backend(self):
         await asyncio.sleep(0.5)
         self.root.transition = kivy.uix.screenmanager.FadeTransition()
-        self.root.current = "facescreen"
+
+        # set the initial screen
+        self.root.current = ScreenConfig.FACE_SCREEN_NAME
+        self.screen_stack.append(self.root.current)
 
         try:
             while True:
 
+                # update screen upon new state
+                # need better way. probably just another state
+                if len(self.ros_set_screen) > 6:
+                    self.root.current = self.ros_set_screen
+                    self.ros_set_screen = ""

```

Comment:
This is dangerous code, any value could get put into self.ros_set_screen and if it is not a screen name the app will crash. Use something like: 
`if self.ros_set_screen and self.get_screen(self.ros_set_screen):`
`    self.root.current = self.ros_set_screen`
`    self.ros_set_screen = None`

the get_screen method is from the screen_wrapper class